File: SWEA/SWEA_D3_1208.py

# 231109

for i in range(10):

    result = -1
    dump = int(input())
    heights = list(map(int, input().split()))

    heights.sort(reverse=True)

    while True:
        if dump == 0 or heights[0] - heights[-1] <= 1:
            result = heights[0] - heights[-1]
            break

        heights[0] -= 1
        heights[-1] += 1
        heights.sort(reverse=True)
        dump -= 1

    print(f"#{i+1} {result}")

File: Implementation/SWEA_D3_1208.java
package SWEA;

import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.Arrays;

// 240130
public class SWEA_D3_1208 {

    public static void main(String[] args) throws Exception {

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        for (int t = 1; t <= 10; t++) {

            int result = 0;
            int dump = Integer.parseInt(br.readLine());

            int[] map = Arrays.stream(br.readLine().split(" "))
                .mapToInt(s -> Integer.parseInt(s))
                .sorted()
                .toArray();

            while (dump != 0) {
                if (map[map.length - 1] == map[0]) {
                    break;
                }

                map[map.length - 1] -= 1;
                map[0] += 1;
                Arrays.sort(map);
                dump -= 1;
            }

            result = map[map.length - 1] - map[0];

            System.out.printf("#%d %d\n", t, result);
        }
    }
}
